frontend/www/scripts/workouts.js

```diff
@@ -37,6 +37,47 @@ function createWorkout() {
     window.location.replace("workout.html");
 }
 
+function filterWorkouts(searchValue, authorFilter, workouts, currentUser) {
+    let workoutAnchors = document.querySelectorAll('.workout');
+    for (let j = 0; j < workouts.length; j++) {
+        // I'm assuming that the order of workout objects matches
+        // the other of the workout anchor elements. They should, given
+        // that I just created them.
+        let workout = workouts[j];
+        let workoutAnchor = workoutAnchors[j];
```

Comment:
These variables should probably be `const` since they're not mutated, but we can change that later.